parsing/src/recipes.ts

```diff
@@ -142,32 +139,138 @@ function getRecipes(
                 products,
                 building,
                 isAlternate: recipe.mDisplayName.includes("Alternate"),
-                isFicsmas: isFicsmas(recipe.mDisplayName)
+                isFicsmas: isFicsmas(recipe.mDisplayName),
+                isPowerGenerator: false
             });
         });
 
-    // Manually add Nuclear waste recipes
-    recipes.push({
-        id: "NuclearWaste",
-        displayName: "Uranium Waste",
-        ingredients: [{ part: 'NuclearFuelRod', amount: 1, perMin: 0.2 }, { part: 'Water', amount: 1200, perMin: 240 }],
-        products: [{ part: "NuclearWaste", amount: 1, perMin: 10 }],
-        building: { name: "nuclearpowerplant", power: 0 },
-        isAlternate: false,
-        isFicsmas: false
-    });
-    recipes.push({
-        id: "PlutoniumWaste",
-        displayName: "Plutonium Waste",
-        ingredients: [{ part: 'PlutoniumFuelRod', amount: 1, perMin: 0.1 }, { part: 'Water', amount: 2400, perMin: 240 }],
-        products: [{ part: "PlutoniumWaste", amount: 1, perMin: 1 }],
-        building: { name: "nuclearpowerplant", power: 0 },
-        isAlternate: false,
-        isFicsmas: false
-    });
+    // // Manually add Nuclear waste recipes
+    // recipes.push({
+    //     id: "NuclearWaste",
+    //     displayName: "Uranium Waste",
+    //     ingredients: [{ part: 'NuclearFuelRod', amount: 1, perMin: 0.2 }, { part: 'Water', amount: 1200, perMin: 240 }],
+    //     products: [{ part: "NuclearWaste", amount: 1, perMin: 10 }],
+    //     building: { name: "nuclearpowerplant", power: 2500 },
+    //     isAlternate: false,
+    //     isFicsmas: false,
+    //     isPowerGenerator: true
+    // });
+    // recipes.push({
+    //     id: "PlutoniumWaste",
+    //     displayName: "Plutonium Waste",
+    //     ingredients: [{ part: 'PlutoniumFuelRod', amount: 1, perMin: 0.1 }, { part: 'Water', amount: 2400, perMin: 240 }],
+    //     products: [{ part: "PlutoniumWaste", amount: 1, perMin: 1 }],
+    //     building: { name: "nuclearpowerplant", power: 2500 },
+    //     isAlternate: false,
+    //     isFicsmas: false,
+    //     isPowerGenerator: true
+    // });
+
+    return recipes.sort((a, b) => a.displayName.localeCompare(b.displayName));
+}
+
+function getPowerGeneratingRecipes(
```

Comment:
I advise you abstract this into it's own file, and then write tests upon it.

Reply:
Will do in next PR.